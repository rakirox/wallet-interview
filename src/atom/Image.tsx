import React, {useState} from 'react';
import {
  StyleSheet,
  Image as BaseImage,
  ImageProps,
  ActivityIndicator,
  View,
} from 'react-native';

export function Image(tProps: ImageProps): JSX.Element {
  const [loading, setLoading] = useState(true);

  return (
    <>
      <BaseImage
        {...tProps}
        style={[styles.image, tProps.style]}
        onLoadEnd={() => setLoading(false)}
      />
      {loading && (
        <View style={styles.absolute}>
          <ActivityIndicator color="rgba(0,0,0,0.5)" />
        </View>
      )}
    </>
  );
}

const styles = StyleSheet.create({
  image: {},
  absolute: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
