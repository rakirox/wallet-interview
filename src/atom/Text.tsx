import React from 'react';
import {StyleSheet, Text as BaseText, TextProps} from 'react-native';

export function Text(tProps: TextProps): JSX.Element {
  return <BaseText {...tProps} style={[styles.text, tProps.style]} />;
}

const styles = StyleSheet.create({
  text: {
    fontFamily: 'avenir',
  },
});
