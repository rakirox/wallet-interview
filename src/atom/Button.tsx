import React from 'react';
import {
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import {Text} from './Text';

type Props = {
  text: string;
  onPress: () => void;
  textStyle?: TextStyle;
} & TouchableOpacityProps;

export function Button({
  text,
  onPress,
  style,
  textStyle,
  ...tProps
}: Props): JSX.Element {
  return (
    <TouchableOpacity
      {...tProps}
      style={[styles.button, style]}
      onPress={onPress}>
      <Text style={[styles.text, textStyle, textStyle]}>{text}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#334FFA',
    borderRadius: 10,
  },
  text: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 22,
    color: 'white',
    alignSelf: 'center',
    paddingVertical: 14,
  },
});
