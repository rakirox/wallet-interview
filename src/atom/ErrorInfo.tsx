import React, {PropsWithChildren} from 'react';
import {View, StyleSheet} from 'react-native';
import {Text} from './Text';

type Props = PropsWithChildren<{
  text: string;
}>;

export function ErrorInfo({text}: Props) {
  return (
    <View style={styles.container}>
      <Text style={styles.errorText}>Error encontrado: {text}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  errorText: {
    color: 'red',
  },
});
