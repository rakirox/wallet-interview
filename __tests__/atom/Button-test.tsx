/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Button} from '../../src/atom/Button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Button - render correctly', () => {
  expect(
    renderer.create(<Button text="Hello Femsa" onPress={jest.fn} />),
  ).toMatchSnapshot();
});
