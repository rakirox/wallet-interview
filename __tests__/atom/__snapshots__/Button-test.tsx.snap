// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "avenir",
        },
        [
          {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 22,
            "paddingVertical": 14,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    Hello Femsa
  </Text>
</View>
`;
